velox/dwio/dwrf/common/CachedBufferedInput.cpp

```diff
@@ -59,193 +79,321 @@ bool CachedBufferedInput::shouldPreload() {
   int32_t numPages = 0;
   for (auto& request : requests_) {
     numPages += bits::roundUp(
-                    std::min<int32_t>(
-                        request.size, CacheInputStream::kDefaultLoadQuantum),
+                    std::min<int32_t>(request.size, FLAGS_cache_load_quantum),
                     MappedMemory::kPageSize) /
         MappedMemory::kPageSize;
   }
   auto cachePages = cache_->incrementCachedPages(0);
   auto maxPages = cache_->maxBytes() / MappedMemory::kPageSize;
   auto allocatedPages = cache_->numAllocated();
   if (numPages < maxPages - allocatedPages) {
+    // There is free space for the read-ahead.
     return true;
   }
   auto prefetchPages = cache_->incrementPrefetchPages(0);
   if (numPages + prefetchPages < cachePages / 2) {
+    // The planned prefetch plus other prefetches are under half the cache.
     return true;
   }
   return false;
 }
 
+namespace {
+
+std::vector<int32_t> makeReadPctBuckets() {
+  return {80, 50, 20, 0};
+}
+
+const static std::vector<int32_t>& readPctBuckets() {
+  static std::vector<int32_t> buckets = makeReadPctBuckets();
```

Comment:
makeReadPctBuckets function seems unnecessary. Why can't you just initialize the vector as buckets = {80, 50, 20, 0};?